pkg/operator/apiservices/sync_apiservices_oauth_apiserver.go

```diff
@@ -1,88 +0,0 @@
-package apiservices
-
-import (
-	"sync"
-
-	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/klog/v2"
-	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
-
-	operatorlistersv1 "github.com/openshift/client-go/operator/listers/operator/v1"
-	"github.com/openshift/library-go/pkg/operator/events"
-)
-
-// APIServicesToManage returns the current list of APIServices managed by this operator.
-// If authOperator.status.managingOAuthAPIServer is false, the empty is returned, otherwise
-// the full list of APIServices.
-type APIServicesToManage struct {
-	authOperatorLister operatorlistersv1.AuthenticationLister
-	all                []*apiregistrationv1.APIService
-}
-
-// NewAPIServicesToManage returns an object that knows how to construct an authoritative list of API services this operator must manage
-func NewAPIServicesToManage(authOperatorLister operatorlistersv1.AuthenticationLister, all []*apiregistrationv1.APIService) *APIServicesToManage {
-	return &APIServicesToManage{
-		authOperatorLister: authOperatorLister,
-		all:                all,
-	}
-}
-
-// GetAPIServicesToManage returns the desired list of API Services that will be managed by this operator
-// Note that at the moment the returned list is dynamic and depends on authOperator.Status.ManagingOAuthAPIServer field
-//
-// TODO: change this function in 4.7 to return initial/full/authoritative list of APIs to manage
-func (a *APIServicesToManage) GetAPIServicesToManage() ([]*apiregistrationv1.APIService, error) {
-	authOperator, err := a.authOperatorLister.Get("cluster")
-	if err != nil {
-		klog.V(4).Infof("unable to determine if auth operator should take OAuth APIs over due to %v", err)
-		return nil, err
-	}
-
-	if !authOperator.Status.ManagingOAuthAPIServer {
```

Comment:
we also need to decide how to depreciate `ManagingOAuthAPIServer` field. 

The situation here is a bit different because we are going to remove it from the API so it will be pruned and removed from the database on an update.

We could remove it in 4.8 and set it to `true` in 4.7 so that a downgrade from 4.7 to 4.6 is smooth.

Reply:
Makes sense

Reply:
we decided to remove it in `4.8` along with the annotation.